src/api/validators/biometrics/blood.oxygen.saturation.validator.ts

```diff
@@ -0,0 +1,218 @@
+import express from 'express';
+import { body, param, validationResult, query } from 'express-validator';
+import { Logger } from '../../../common/logger';
+import { Helper } from '../../../common/helper';
+import { BloodOxygenSaturationDomainModel } from '../../../domain.types/biometrics/blood.oxygen.saturation/blood.oxygen.saturation.domain.model';
+import { BloodOxygenSaturationSearchFilters } from '../../../domain.types/biometrics/blood.oxygen.saturation/blood.oxygen.saturation.search.types';
+
+///////////////////////////////////////////////////////////////////////////////////////
+
+export class BloodOxygenSaturationValidator {
+
+    static getDomainModel = (request: express.Request): BloodOxygenSaturationDomainModel => {
+
+        const BloodOxygenSaturationModel: BloodOxygenSaturationDomainModel = {
+            PatientUserId         : request.body.PatientUserId,
+            BloodOxygenSaturation : request.body.BloodOxygenSaturation,
+            Unit                  : request.body.Unit,
+            RecordDate            : request.body.RecordDate ?? null,
+            RecordedByUserId      : request.body.RecordedByUserId ?? null,
+
+        };
+
+        return BloodOxygenSaturationModel;
+    };
+
+    static create = async (request: express.Request): Promise<BloodOxygenSaturationDomainModel> => {
+        await BloodOxygenSaturationValidator.validateBody(request);
+        return BloodOxygenSaturationValidator.getDomainModel(request);
+    };
+
+    static getById = async (request: express.Request): Promise<string> => {
+        return await BloodOxygenSaturationValidator.getParamId(request);
+    };
+
+    static getByPatientUserId = async (request: express.Request): Promise<string> => {
+
+        await param('PatientUserId').trim()
+            .escape()
+            .isUUID()
+            .run(request);
+
+        const result = validationResult(request);
+
+        if (!result.isEmpty()) {
+            Helper.handleValidationError(result);
+        }
+
+        return request.params.PatientUserId;
+    };
+
+    static getByPersonId = async (request: express.Request): Promise<string> => {
+
+        await param('personId').trim()
+            .escape()
+            .isUUID()
+            .run(request);
+
+        const result = validationResult(request);
+
+        if (!result.isEmpty()) {
+            Helper.handleValidationError(result);
+        }
+
+        return request.params.personId;
+    };
+
+    static delete = async (request: express.Request): Promise<string> => {
+        return await BloodOxygenSaturationValidator.getParamId(request);
+    };
+
+    static search = async (request: express.Request): Promise<BloodOxygenSaturationSearchFilters> => {
+
+        await query('PatientUserId').optional()
```

Comment:
Anything we add to request params and query params such as ids should be camelCase.
i.e. request.query.**patientUserId**.